pkg/internal/config/config.go

```diff
@@ -30,16 +31,47 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+const (
+	StorageClassParamName = "storageClass"
+	VMITimeoutParamName   = "vmiTimeout"
+)
+
+const (
+	VMITimeoutDefault = 3 * time.Minute
+)
+
+var (
+	ErrInvalidVMITimeout = errors.New("invalid VMI timeout")
+)
+
 type Config struct {
-	PodName string
-	PodUID  string
+	PodName      string
+	PodUID       string
+	StorageClass string
+	VMITimeout   time.Duration
 }
 
 func New(baseConfig kconfig.Config) (Config, error) {
 	newConfig := Config{
-		PodName: baseConfig.PodName,
-		PodUID:  baseConfig.PodUID,
+		PodName:      baseConfig.PodName,
+		PodUID:       baseConfig.PodUID,
+		StorageClass: baseConfig.Params[StorageClassParamName],
```

Comment:
I think since storageclass is also optional you shouldnt set it here, but you should do it in the set optional params.. I know that in the current way it will also be empty if it wasnt set but code wise its a bit confusing.